File: backend/go/client/nytimes_client.go
package client

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"

	"github.com/osucoding/simple-weather-app/model"
	"github.com/osucoding/simple-weather-app/util"
)

func GetNews(sectionValue string) (model.NyTimesResponse, error) {
	nyTimesConfig := util.GetNyTimesConfig()

	url := fmt.Sprintf("%s%s.json?api-key=%s", nyTimesConfig.Nytimes.Uri, sectionValue, nyTimesConfig.Nytimes.ApiKey)

	response, err := http.Get(url)
	if err != nil {
		return model.NyTimesResponse{}, err
	}

	body, err := io.ReadAll(response.Body)
	if err != nil {
		return model.NyTimesResponse{}, err
	}

	var news model.NyTimesResponse
	if err := json.Unmarshal(body, &news); err != nil {
		return model.NyTimesResponse{}, err
	}

	return news, nil
}

File: backend/python/client/nytimes_client.py
import requests

from model.ny_times import NyTimesResponse, ResultsItem
from util.resources import get_ny_times_api_resource


def get_news(section_value: str) -> NyTimesResponse:
    # get APIResource
    ny_times_api_config = get_ny_times_api_resource()
    params = {
        'api-key': ny_times_api_config.api_key
    }

    base_url = f'{ny_times_api_config.uri}{section_value}.json'

    response = requests.get(base_url, params)

    if response.status_code == 200:
        data = response.json()
        return _parse_response(data)
    else:
        response.raise_for_status()

def _parse_response(data: dict) -> NyTimesResponse:
    return NyTimesResponse(**data)